comixed-tasks/src/main/java/org/comixedproject/task/model/DeleteComicsWorkerTask.java

```diff
@@ -51,15 +60,20 @@ protected String createDescription() {
   public void startTask() throws WorkerTaskException {
     for (int index = 0; index < this.comicIds.size(); index++) {
       final Long id = this.comicIds.get(index);
-      final Comic comic = this.comicRepository.getById(id);
+      Comic comic;
+      try {
+        comic = this.comicService.getComic(id);
+      } catch (ComicException error) {
+        throw new WorkerTaskException("failed to load comic", error);
+      }
```

Comment:
an enhanced loop would be more readable

Reply:
Yes, but I believe we'd introduce the ConcurrentModificationException issue with JPA.

Reply:
Ok